### PR TITLE
dispatch search on location change

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds a middleware to dispatch search (http request) on
LOCATION_CHANGE so that url and content always synced correctly. Now
all search components dispatches action that pushes the search query to
location state then later handled by the middleware

* Refactors search component tests (mocking the redux parts)

* Removes unnecesary holdingpen/SearchPage and tests for it

### CODE REVIEW

File: src/middlewares/searchDispatcher.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { LOCATION_CHANGE } from 'react-router-redux';
+import { searchForCurrentLocation } from '../actions/search';
+
+function getLocationFromRootState(state) {
+  const { router } = state;
+  const { location } = router;
```
Comment (drjova) [inline src/middlewares/searchDispatcher.js]: why not 

```js
const { router: {location}  } = state;
```

File: src/actions/__tests__/search.test.js [hunk 2]
```diff
@@ -28,152 +28,182 @@ const stateWithoutScopeQuery = {
...
+          location: { pathname: '/test', search: '?size=10&q=test' },
+        },
+      });
+      mockHttp.onGet('/test?size=10&q=test').networkError();
+
+      await store.dispatch(searchForCurrentLocation());
+
+      const expectedActions = [
+        { type: types.SEARCH_REQUEST },
+        { type: types.SEARCH_ERROR, payload: undefined },
```
Comment (drjova) [inline src/actions/__tests__/search.test.js]: shall we also fix the ``undefined``?

File: src/actions/__tests__/search.test.js [hunk 3]
```diff
@@ -28,152 +28,182 @@ const stateWithoutScopeQuery = {
...
-      { type: types.SEARCH_ERROR, payload: undefined },
-    ];
-
-    const store = getStoreWithState(stateWithScopeQuery);
-    await store.dispatch(search({ q: 'test' }));
-    expect(store.getActions()).toEqual(expectedActions);
-    done();
+  describe('pushQueryToLocation', () => {
+    it('pushes new location url to history', async done => {
+      const expectedUrl = '/test?size=10&q=test';
```
Comment (drjova) [inline src/actions/__tests__/search.test.js]: shouldn't the expected be ```size=10&page=1&q=test``?

File: src/middlewares/searchDispatcher.js (hunk 1 above)
Comment (harunurhan) [reply src/middlewares/searchDispatcher.js]: just personal preference :), but I'll do that

File: src/actions/__tests__/search.test.js (hunk 2 above)
Comment (harunurhan) [reply src/actions/__tests__/search.test.js]: well... we have to fix error handling for this probably. It requires changing initialState of error and dispatching search error (`searchError(error.response && error.response.data)`) and perhaps more. Let's fix that all together for all reducers/actions

File: src/actions/__tests__/search.test.js (hunk 3 above)
Comment (harunurhan) [reply src/actions/__tests__/search.test.js]: why do you assume that? 

`page=1` is set only if a new query is made and there is already `page` param set in the current location.

**Pull request merged**